### PR TITLE
multi-thread pushout

### PR DESCRIPTION
**Pull request by xmacan**
Please test and comment

### CODE REVIEW

File: cli/push_out_hosts.php [hunk 1]
```diff
@@ -23,24 +23,49 @@
...
+/* mandatory parameters */
+$start_time        = false;
+$end_time          = false;
+
+/* optional parameters for host selection */
 $debug            = false;
-$host_id          = 0;
-$host_template_id = 0;
-$data_template_id = 0;
+$host_id          = false;
```
[the remark this answers is not in the document]
Comment (xmacan) [reply cli/push_out_hosts.php:60]: ok, I will fix it soon. Thank you

**Comment by TheWitness** [comment]
@xmacan, please update the changelog too.  Good work!

File: cli/push_out_hosts.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (xmacan) [reply cli/push_out_hosts.php:60]: Hmm, it seems that all cli scripts are using false now. I would like to keep this for now. Maybe change it later everywhere at once

**Comment by xmacan** [comment]
Pushout for version 1.2.x needs a few minor changes. Should I prepare it for 1.2.x or not?

**Comment by TheWitness** [comment]
Might as well, it you are up to it.  We are doing another release likely late may.

**Pull request merged**